spec/services/api/training_periods/teacher_status_spec.rb

```diff
@@ -0,0 +1,79 @@
+RSpec.describe API::TrainingPeriods::TeacherStatus do
+  let(:service) { described_class.new(training_period:, teacher:) }
+  let(:teacher) { training_period.trainee.teacher }
+
+  describe "#status" do
+    subject { service.status }
+
+    context "when training period has started and not set to finish" do
+      let(:training_period) { FactoryBot.build(:training_period, :ongoing) }
+
+      it { is_expected.to eq(:active) }
+      it { expect(service).to be_active }
+      it { expect(service).not_to be_joining }
+      it { expect(service).not_to be_leaving }
+      it { expect(service).not_to be_left }
+    end
+
+    context "when 'teacher.mentor_became_ineligible_for_funding_on' is set" do
```

Comment:
Can we maybe call out what this means as I don't think its particularly obvious for this and the `finished_induction_period`, maybe:

`context "when 'teacher.mentor_became_ineligible_for_funding_on' is set (indicating the mentor has completed training)"`

`context "when 'finished_induction_period.finished_on' is set (indicating the ECT has completed training)"`